syntax = "proto3";

import "google/protobuf/empty.proto";
import "google/protobuf/duration.proto";
import "google/protobuf/timestamp.proto";

package com.centreon.engine;

service Engine {
  rpc GetVersion(google.protobuf.Empty) returns (Version) {}
  rpc GetStats(google.protobuf.Empty) returns (Stats) {}

  rpc ProcessServiceCheckResult(Check) returns (CommandSuccess) {}
  rpc ProcessHostCheckResult(Check) returns (CommandSuccess) {}

  rpc NewThresholdsFile(ThresholdsFile) returns (CommandSuccess) {}
}

message CommandSuccess {
  bool value = 1;
}

message Check {
  google.protobuf.Timestamp check_time = 1;
  string host_name = 2;
  string svc_desc = 3;
  string output = 4;
  uint32 code = 5;
}

message Version {
  int32 major = 1;
  int32 minor = 2;
  int32 patch = 3;
}

message ProgramStatus {
  uint32 modified_host_attributes = 1;
  uint32 modified_service_attributes = 2;
  uint32 pid = 3;
  google.protobuf.Timestamp program_start = 4;
  google.protobuf.Timestamp last_command_check = 5;
  google.protobuf.Timestamp last_log_rotation = 6;
  bool enable_notifications = 7;
  bool active_service_checks_enabled = 8;
  bool passive_service_checks_enabled = 9;
  bool active_host_checks_enabled = 10;
  bool passive_host_checks_enabled = 11;
  bool enable_event_handlers = 12;
  bool obsess_over_services = 13;
  bool obsess_over_hosts = 14;
  bool check_service_freshness = 15;
  bool check_host_freshness = 16;
  bool enable_flap_detection = 17;
  bool process_performance_data = 18;
  string global_host_event_handler= 19;
  string global_service_event_handler = 20;
  uint64 next_comment_id = 21;
  uint64 next_event_id = 22;
  uint64 next_problem_id = 23;
  uint64 next_notification_id = 24;
  uint32 total_external_command_buffer_slots = 25;
  uint32 used_external_command_buffer_slots = 26;
  uint32 high_external_command_buffer_slots = 27;
  repeated uint32 active_scheduled_host_check_stats = 28;
  repeated uint32 active_ondemand_host_check_stats = 29;
  repeated uint32 passive_host_check_stats = 30;
  repeated uint32 active_scheduled_service_check_stats = 31;
  repeated uint32 active_ondemand_service_check_stats = 32;
  repeated uint32 passive_service_check_stats = 33;
  repeated uint32 cached_host_check_stats = 34;
  repeated uint32 cached_service_check_stats = 35;
  repeated uint32 external_command_stats = 36;
  repeated uint32 parallel_host_check_stats = 37;
  repeated uint32 serial_host_check_stats = 38;
}

message ExtCmdBuffer {
 uint32 used = 1;
 uint32 high = 2;
 uint32 total = 3;
}

message Stats {
  ProgramStatus program_status = 1;
  ExtCmdBuffer buffer = 2;
}

message ThresholdsFile {
  string filename = 1;
}
